File: Fitness/Resources/Storage.m
//
//  Storage.m
//  Fitness
//
//  Created by Mani on 25.04.15.
//  Copyright (c) 2015 Mani. All rights reserved.
//

#import "Storage.h"

#define KEY "trainings"

@implementation Storage

#pragma mark Methods
+ (NSMutableArray *)trainings {
    NSData *data = [[NSUserDefaults standardUserDefaults] objectForKey:@KEY];
    if(data == nil)
        return [NSMutableArray array];
    
    return [NSKeyedUnarchiver unarchiveObjectWithData:data];
}

+ (BOOL)saveTrainings:(NSArray *)trainings {
    NSData *data = [NSKeyedArchiver archivedDataWithRootObject:trainings];
    
    [[NSUserDefaults standardUserDefaults] setObject:data forKey:@KEY];
    return [[NSUserDefaults standardUserDefaults] synchronize];
}

@end

File: Fitness/Core/Storage.swift
//
//  Storage.swift
//  Fitness
//
//  Created by Manuel Stampfl on 26.06.15.
//  Copyright (c) 2015 mani1337. All rights reserved.
//

import Foundation

private var defaults = NSUserDefaults(suiteName: "group.at.mani1337.Fitness")

private func readObjectForKey(key: String) -> AnyObject? {
    if let object = defaults?.objectForKey(key) as? NSData {
        return NSKeyedUnarchiver.unarchiveObjectWithData(object)
    }
    
    return nil
}

private func writeObjectForKey(key: String, object: AnyObject) -> Bool {
    if defaults == nil {
        return false
    }
    
    defaults!.setObject(NSKeyedArchiver.archivedDataWithRootObject(object), forKey: key)
    return defaults!.synchronize()
}

class Storage {
    class var trainings: [Training] {
        get {
            if let data = readObjectForKey("trainings") as? [Training] {
                return data
            }
            
            return []
        } set {
            writeObjectForKey("trainings", newValue)
        }
    }
}